### PR TITLE
Fix select dynamic custom attributes

### PR DESCRIPTION
**Pull request by klas**
Select dynamic custom attributes are not working in Free mode, they are always empty like data-sth="". Also interface is behaving wrong, first attribute is initially shown then on toggle first one is hidden and 2-6 are shown. This pull fixes both problems.

### CODE REVIEW

**Comment by klas** [comment]
Also fixed it for construction mode after fixes so now they should both work

File: plugins/cck_field/select_dynamic/tmpl/edit.php [hunk 1]
```diff
@@ -79,7 +79,7 @@
 	$('#json_options2_table, #json_options2_name, #json_options2_value, #json_options2_where, #json_options2_orderby, #json_options2_limit, #json_options2_attr1, #blank_li').isVisibleWhen('bool2','0');
```
Comment (olivier-at-octopoos) [inline plugins/cck_field/select_dynamic/tmpl/edit.php]: Klas,

$('#json_options2_table, #json_options2_name, #json_options2_value, #json_options2_where, #json_options2_orderby, #json_options2_limit, #blank_li').isVisibleWhen('bool2','0');

This line should be like this, without the '#json_options2_attr1' in the list of fields to show if #bool2=0. This is what really causes the incorrect behavior with the custom attributes toggle.

Comment (klas) [inline plugins/cck_field/select_dynamic/tmpl/edit.php]: Could be I was fixing symptoms instead the cause, I'll take another look

**Comment by sebastienheraud** [comment]
Hi guys.
Custom attributes were working fine in Free mode.
The right way to use them was to write the query like that:
"..., catid AS attr1, featured AS attr2"
I'm okay to have an alternative way to declare them (without anything in the query) but we need to make sure that changes don't break anything..
Here is what we I've merged: https://github.com/Octopoos/SEBLOD/commit/07318c2041bde19e73cd3bb8bb25fe6ad5e72303
The "edit.php" has been mergedas well.

Thank you!
Saba

**Comment by klas** [comment]
Sorry, they were not working right, there were 2 things that were broken:

a) interface with toggle closing first attribute
b) empty values if you put e.g. select abc, cde in query and then have something abc and something2 cde in the cutom attributes options, attaching real life examle I tested this with (screenshot from my fixed version)
belmondo_-_pocitnice _potovanja _hoteli _letalske_karte _lastminute_-_administration_-_field_small_small_ _edit_ _small_small_-_2015-07-02_17 29 17

**Comment by olivier-at-octopoos** [comment]
Klas,

For me point B was not an issue but just an incorrect use of the feature.

The good point is now we have two ways to use the custom attributes !!!

Cheers

**Comment by klas** [comment]
@olivier-at-octopoos label of the custom attributes says Custom attr / Column with corresponding 2 boxes, not sure how to understand this differently than as "custom attribute name" for first box and "database column to use for attribute value" in second. If you see my screenshot I had both those columns also in the query and it wasn't working. Would be good if there would be some  documentation how to use this feature...

**Pull request closed without merging**